File: private/apigen/example/project.go
// Copyright (C) 2023 Storj Labs, Inc.
// See LICENSE for copying information.

package example

import (
	"storj.io/common/uuid"
)

// Project contains project information.
type Project struct {
	ID        uuid.UUID `json:"id"`
	OwnerName string    `json:"ownerName"`
}

File: satellite/admin/ui/src/types/project.ts
// Copyright (C) 2025 Storj Labs, Inc.
// See LICENSE for copying information.

export enum ProjectStatus {
    Disabled = 0,
    Active = 1,
    PendingDeletion = 2,
}

export interface MembersCursor {
    search: string;
    limit: number;
    page: number;
    order: ProjectMemberOrderBy;
    orderDirection: ProjectMemberOrderDirection;
}

export enum ProjectMemberOrderBy {
    EMAIL = 1,
    DATE,
}

export enum ProjectMemberOrderDirection {
    ASC = 1,
    DESC = 2,
}

export enum ProjectMemberRole {
    ADMIN = 0,
    MEMBER = 1,
}
